datachannel.go

```diff
@@ -151,16 +154,16 @@ func (d *DataChannel) open(sctpTransport *SCTPTransport) error {
 		LoggerFactory:        d.api.settingEngine.LoggerFactory,
 	}
 
-	if d.id == nil {
+	if d.ID() == nil {
 		generatedID, err := d.sctpTransport.generateDataChannelID(d.sctpTransport.dtlsTransport.role())
 		if err != nil {
 			return err
 		}
 
-		d.id = &generatedID
+		d.id.Store(&generatedID)
```

Comment:
Data race of `id` is just hidden by using atomic, but concurrent call silently causes duplicate of ID.
```go
// routine 1
	if d.ID() == nil {
		generatedID, err := d.sctpTransport.generateDataChannelID(d.sctpTransport.dtlsTransport.role())
// routine 2
	if d.ID() == nil {
		generatedID, err := d.sctpTransport.generateDataChannelID(d.sctpTransport.dtlsTransport.role())
// routine 1
		d.id.Store(&generatedID)
// routine 2
		d.id.Store(&generatedID)

Reply:
oof didn't think about that at all, and what you are doing is making `generateDataChannelID` only able to running one at a time.

Reply:
Will close PR and approve yours, thanks for catching that!